lib/fog/orchestration/util/recursive_hot_file_loader.rb

```diff
@@ -0,0 +1,227 @@
+require 'yaml'
+require 'open-uri'
+require 'objspace'
+require 'fog/core'
+require 'set'
+
+module Fog
+  module Orchestration
+    module Util
+      #
+      # Resolve get_file resources found in a HOT template populating
+      #  a files Hash conforming to Heat Specs
+      #  https://developer.openstack.org/api-ref/orchestration/v1/index.html?expanded=create-stack-detail#stacks
+      #
+      # This implementation just process nested templates but not resource
+      #  registries.
+      class RecursiveHotFileLoader
```

Comment:
The logic how to traverse nodes and load files seems over complicated. I have hard time to follow. Can you add comments to each method? We can remove some of them if they turn out to be straightforward.

Reply:
@bzwei agree but:

- my first implementation was simpler
- then I preferred to mimic the more stable python implementation